crates/ffs/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+//! ffs - the Friendly Filesystem
+//!
+//! Wraps std::fs (or optionally, tokio::fs) in functions with identical
+//! signatures such that error messages include extra context, in particular
+//! the path used in the function call.
+//!
+//! This helps with "file not found" errors. Without ffs, the error would be:
+//!
+//!     Error: No such file or directory (os error 2)
+//!
+//! and with ffs, the error becomes:
+//!
+//!     ffs::IoError at path '/foo/bar': No such file or directory (os error 2)
+
+mod io_error;
+
+pub use crate::io_error::{IoError, IoResult};
+
+macro_rules! impl_ffs {
+    ( $( fn $name:ident (path $(, $arg:ident : $arg_ty:ty)* ) -> $output:ty ; )* ) => {
+
+        $(
+            pub async fn $name<P: Clone + AsRef<std::path::Path>>(path: P $(, $arg : $arg_ty)*) -> IoResult<$output> {
+                let err_path = path.as_ref().to_owned();
+                let mapper = move |e| IoError::new(e, err_path);
+
+                #[cfg(feature = "tokio")]
```

Comment:
Just noting there are valid reasons where you need to use std and tokio in the same project but I guess you can still use them without this macro right?

Reply:
Yeah, `ffs::sync` just re-exports all the sync methods from `std::fs`.

Well, it re-exports the ones that I bothered to add to the implementation anyway.